code/ui/scoreboard/Scoreboard.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Sandbox;
+using Sandbox.UI;
+using System;
+using System.Collections.Generic;
+
+namespace Cinema.UI;
+
+public partial class Scoreboard : Panel, IMenuScreen
+{
+    public static Scoreboard Instance { get; set; }
+    public string Name => Game.Server.ServerTitle;
+    public bool IsOpen { get; protected set; }
+    public string VisibleClass => IsOpen ? "visible" : "";
+    public IReadOnlyCollection<IClient> Clients => Game.Clients;
+
+    public Scoreboard()
+    {
+        Instance = this;
+    }
+    public bool Open()
+    {
+        IsOpen = true;
+
+        return true;
+    }
+    public void Close()
+    {
+        IsOpen = false;
+    }
+    protected override int BuildHash()
+    {
+        var queueHash = 11;
+
+        return HashCode.Combine(IsOpen, queueHash, Clients, Clients.Count);
```

Comment:
Just throwing `Clients` in here won't actually work the way you intend it to.

You'd have to look through each client and compound a hash with the properties you actually care about.

Reply:
I'm thinking about updating the hash function with the code snippet below. Does it look alright? 

```
protected override int BuildHash()
{
    var clientHash = 11;

    foreach(var client in Clients)
    {
        clientHash = clientHash * 31 + client.GetHashCode();
    }

    return HashCode.Combine(IsOpen, Clients.Count, clientHash);
}